### PR TITLE
autogtp: fixing game end in validation job

### PR DESCRIPTION
**Pull request by marcocalignano**

### CODE REVIEW

**Comment by roy7** [comment]
Once this is merged and we roll it out so no more bad results will be coming in, I can try to do some queries on the database to fix the errors manually.

**Comment by marcocalignano** [comment]
WE need to test it you found the problem with the three passes can you reproduce it?

**Comment by zediir** [comment]
Could you use the random_seed parameter to test the exact same game if find a problematic one?

**Comment by marcocalignano** [comment]
@zediir I guess not it is only a seed but I am not sure

**Comment by roy7** [comment]
Speaking of seeds, is it fine that both black and white use the same seed?

**Comment by zediir** [comment]
I would think it's fine because because they use different networks. Even if they used the same networks I guess the result would just be the same as if running on a single leelaz instance.

**Comment by roy7** [comment]
Spot on @zediir. Thanks.

**Comment by marcocalignano** [comment]
I tested and it is working.

File: autogtp/Game.h [hunk 1]
```diff
@@ -42,6 +42,7 @@ class Game : QProcess {
     QString getMove() const { return m_moveDone; }
     QString getFile() const { return m_fileName; }
     bool setMove(const QString& m);
+    bool checkGameEnd() { return (m_resignation || m_passes > 1 || m_moveNum > (19 * 19 * 2)); }
```
Comment (sethtroisi) [inline autogtp/Game.h]: this is longer than 80 characters, consider splitting onto three lines (and potentially moving to .cpp)

File: autogtp/Job.cpp [hunk 2]
```diff
@@ -106,6 +106,9 @@ Result ValidationJob::execute(){
            return res;
        }
        first.readMove();
+       if(first.checkGameEnd()) {
```
Comment (sethtroisi) [inline autogtp/Job.cpp]: gcp prefers "if (first.checkGameEnd()) {"

see https://github.com/gcp/leela-zero/commit/bcf0b2caa1c1918e686da13f9f34b6899559fd9e

**Comment by marcocalignano** [comment]
IMHO the 80 char rule is mad. Everyone participating in this project has a decent GPU hence has a decent monitor with a decent resolution with **more** than 80 chars per line. But because of this "rule" the compiler now has to make a function call any time `checkGameEnd()` is called instead of just inline that return statement.

**Comment by evanroberts85** [comment]
Some people may be using their mobile phone to monitor remotely?

**Comment by Prillan** [comment]
What's wrong with writing:
```
bool checkGameEnd() { 
    return (m_resignation || m_passes > 1 || m_moveNum > (19 * 19 * 2));
}
```
I'm pretty sure that's what @sethtroisi was talking about.
> consider splitting onto three lines

**Comment by marcocalignano** [comment]
@Prillan Oh well I did my changes already.

**Comment by gcp** [comment]
I'm using 2 console windows side by side and keeping the line length under control helps a lot with that.

Also it's handy if you're editing over ssh from a cellphone :-)

**Comment by marcocalignano** [comment]
@roy7 So now that gcp merged the bug fix you can set the resign rate like it was before and force the client min version to 10 so the people that use the branch next get up to date with the bugfixes

**Comment by roy7** [comment]
Yes I can. So it's 10 and not 9? Required version currently is 8.

**Comment by marcocalignano** [comment]
@roy7 yes the 10 version has the bugfixes the 9 does not!

**Comment by roy7** [comment]
Version 10 now required.

**Pull request merged**